### DISCUSSION TITLE
How to obtain data from getSync/Async methods?

### DISCUSSION THREAD

**Discussion by TerrizX** [spotify-web-api-java/spotify-web-api-java#272] [Q&A] [answered]
For instance, I want to obtain the uri of a Spotify track and put it in another method as a String value, however I'm lost on how I'd go about doing that. I tried experimenting with SharedPreferences to get the value but getString method wasn't working. I was just wondering if there's a simpler way to getting say track.getUri (or any) in another method from the Async/Sync method. Any assistance would be greatly appreciated.

The code so far:

```
private static final String accessToken = "...";
    private static final String id = "01iyCAUm8EvOFqVWYJ3dVX";
    public static SharedPreferences.Editor editor;






    private static final SpotifyApi spotifyApi = new SpotifyApi.Builder()
            .setAccessToken(accessToken)
            .build();
    private static final GetTrackRequest getTrackRequest = spotifyApi.getTrack(id)
//          .market(CountryCode.SE)
            .build();

    public static void getTrack_Sync() {
        try {
            final Track track = getTrackRequest.execute();

            System.out.println("Name: " + track.getName());
        } catch (IOException | SpotifyWebApiException | ParseException e) {
            System.out.println("Error: " + e.getMessage());
        }
    }



    @RequiresApi(api = Build.VERSION_CODES.N)
    public void getTrack_Async() {
        try {
            final CompletableFuture<Track> trackFuture = getTrackRequest.executeAsync();

            // Thread free to do other tasks...


            // Example Only. Never block in production code.
            final Track track = trackFuture.join();

            String uri = track.getUri();
            editor = getSharedPreferences("uri", 0).edit();
            editor.putString("uri", uri);
            editor.commit();
            editor.apply();




            System.out.println("Name: " + track.getUri());

        } catch (CompletionException e) {
            System.out.println("Error: " + e.getCause().getMessage());
        } catch (CancellationException e) {
            System.out.println("Async operation cancelled.");
        }
    }

    public void go() {
        getTrack_Async();
      //  String value = editor.getString("uri", )
    }
```

**Comment by dargmuesli** [comment, marked as answer]
Hey, thanks for the question. Never publish private credentials such as access tokens or client credentials. Also, you can use triple ticks (```) for multiline code. I've removed the access token and formatted the code properly for you.

Regarding your question: please read the very first sentence of this project's readme. This project is concerned with the functionality of the API wrapper. Best practices of usage in your personal projects are subject to be asked on stackoverflow.com, as mentioned in the [CONTRIBUTING.md](https://github.com/spotify-web-api-java/spotify-web-api-java/blob/d3dfa254aac3082e1ad055c629ebf88ccfb414f2/CONTRIBUTING.md).
